mne_qt_browser/_pg_figure.py

```diff
@@ -2209,13 +2216,19 @@ def _init_ui(self):
         layout.addWidget(QLabel('Start:'))
         self.start_bx = QDoubleSpinBox()
         self.start_bx.setDecimals(time_decimals)
-        self.start_bx.editingFinished.connect(self._start_changed)
+        self.start_bx.setMinimum(0)
+        self.start_bx.setMaximum(self.mne.inst.times[-1] - 1 / self.mne.info["sfreq"])
+        self.start_bx.setSingleStep(0.05)
+        self.start_bx.valueChanged.connect(self._start_changed)
         layout.addWidget(self.start_bx)
 
         layout.addWidget(QLabel('Stop:'))
         self.stop_bx = QDoubleSpinBox()
         self.stop_bx.setDecimals(time_decimals)
-        self.stop_bx.editingFinished.connect(self._stop_changed)
+        self.stop_bx.setMinimum(1 / self.mne.info["sfreq"])
+        self.stop_bx.setMaximum(self.mne.inst.times[-1])
+        self.stop_bx.setSingleStep(0.05)
+        self.stop_bx.valueChanged.connect(self._stop_changed)
```

Comment:
The spinBox were free to do whatever. Now they are bound to sensible min/max/step; and the selected region updates on `valueChanged` instead of on `editingFinised` which gives better feedback to the user.

Also, the spinBox are disabled/reset when the *selected* annotation is removed and re-enabled on an annotation selection.